src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+            getGoodReads()
+                .mapPartitionsWithIndex((partIdx,readItr) ->
+                    new ClusterKmerizer(
+                        partIdx,
+                        new BreakpointClusterer(
+                                new SplitReadDetector(),
+                                new DiscordantPairDetector(),
+                                readItr),
+                        killSet.value()),
+                    false)
+                .mapToPair(tupl -> tupl)
+                .combineByKey(breakId -> { Set<Long> coll = new HashSet<>(4); coll.add(breakId); return coll; },
+                        (coll,breakId) -> { coll.add(breakId); return coll; },
+                        (coll1,coll2) -> { coll1.addAll(coll2); return coll1; })
+                .collect();
+
+        // turn the list of interesting Kmers and the set of putative breakpoints to which they belong into a map
+        int nKmers = readPullingKmers.size();
+        final Map<Kmer,List<Long>> readPullingMap = new HashMap<>(5*nKmers/3);
+        for ( Tuple2<Kmer,Set<Long>> tupl : readPullingKmers )
+            readPullingMap.put(tupl._1,new ArrayList<>(tupl._2));
+
+        return readPullingMap;
+    }
+
+    // writes a stream of reads to a FASTQ file
+    private static void writeFastq( Long breakpointId, Iterator<GATKRead> readItr, String outputDir )
+    {
+        File fastqName = new File(outputDir, "assembly" + breakpointId + ".fastq");
+        try
+        {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(fastqName));
+            while ( readItr.hasNext() )
+            {
+                GATKRead read = readItr.next();
+                writer.write("@");
+                writer.write(read.getName());
+                writer.write(read.isFirstOfPair()?":1":":2");
+                writer.newLine();
+                writer.write(read.getBasesString());
+                writer.newLine();
+                writer.write('+');
+                writer.newLine();
+                for ( byte b : read.getBaseQualities() )
+                    writer.write(33+b);
+                writer.newLine();
+            }
+            writer.close();
+        }
+        catch ( IOException e )
+        {
+            throw new GATKException("Can't write "+fastqName+'.',e);
+        }
+    }
+
+    private static final class EventLocus implements Comparable<EventLocus>
+    {
+        private final int locusStart;
+        private final int locusEnd;
+        private long id = 0; // used as a uniquifier
+
+        public EventLocus( final int someLocusStart, final int intervalWidth )
+        {
+            locusStart = someLocusStart;
+            locusEnd = someLocusStart + intervalWidth;
+        }
+
+        public int getLocusStart() { return locusStart; }
+        public int getLocusEnd() { return locusEnd; }
+
+        public EventLocus makeUnique( long someId )
+        {
+            if ( id == 0 )
+                id = someId;
```

Comment:
It's uncomfortable that something that is used as a key in a maps is explicitly mutable